test/python/transpiler/test_clifford_synthesis_plugin.py

```diff
@@ -0,0 +1,204 @@
+
+    def _create_unitaries(self):
+        """
+        Creates unitary gates of varying sizes, some of which can be represented
+        as Cliffords and some of which cannot.
+        """
+
+        # clifford unitary over 1 qubit
+        c1 = QuantumCircuit(1)
+        c1.h(0)
+        c1.s(0)
+        self.uc1 = UnitaryGate(Operator(c1).data)
+
+        # clifford unitary over 2 qubits
+        c2 = QuantumCircuit(2)
+        c2.h(0)
+        c2.rz(math.pi / 4, 1)
+        c2.rz(math.pi / 4, 1)
+        c2.sdg(1)
+        self.uc2 = UnitaryGate(Operator(c2).data)
+
+        # clifford unitary over 3 qubits
+        c3 = QuantumCircuit(3)
+        c3.h(0)
+        c3.cx(0, 1)
+        c3.cx(1, 2)
+        c3.rx(math.pi / 2, 2)
+        self.uc3 = UnitaryGate(Operator(c3).data)
+
+        # non-clifford unitary over 1 qubit
+        n1 = QuantumCircuit(1)
+        n1.h(0)
+        n1.rz(0.8, 0)
+        self.un1 = UnitaryGate(Operator(n1).data)
+
+        # non-clifford unitary over 2 qubits
+        n2 = QuantumCircuit(2)
+        n2.h(0)
+        n2.rz(math.pi / 4, 1)
+        n2.sdg(1)
+        self.un2 = UnitaryGate(Operator(n2).data)
+
+        # non-clifford unitary over 3 qubits
+        n3 = QuantumCircuit(3)
+        n3.h(0)
+        n3.cx(0, 1)
+        n3.cx(1, 2)
+        n3.rx(0.8, 2)
+        self.un3 = UnitaryGate(Operator(n3).data)
+
+    def test_plugin(self):
+        """Test running CliffordUnitarySynthesis plugin directly."""
+
+        plugin = CliffordUnitarySynthesis()
+        clifford_gate_names = set(get_clifford_gate_names())
+
+        with self.subTest(msg="uc1"):
+            mat = self.uc1.to_matrix()
+            out = plugin.run(mat)
+            self.assertLessEqual(set(out.count_ops()), clifford_gate_names)
+            self.assertEqual(Operator(dag_to_circuit(out)), Operator(mat))
+
+        with self.subTest(msg="uc2"):
+            mat = self.uc2.to_matrix()
+            out = plugin.run(mat)
+            self.assertLessEqual(set(out.count_ops()), clifford_gate_names)
+            self.assertEqual(Operator(dag_to_circuit(out)), Operator(mat))
+
+        with self.subTest(msg="uc3"):
+            mat = self.uc3.to_matrix()
+            out = plugin.run(mat)
+            self.assertLessEqual(set(out.count_ops()), clifford_gate_names)
+            self.assertEqual(Operator(dag_to_circuit(out)), Operator(mat))
+
+        with self.subTest(msg="un1"):
+            out = plugin.run(self.un1.to_matrix())
+            self.assertIsNone(out)
+
+        with self.subTest(msg="un2"):
+            out = plugin.run(self.un2.to_matrix())
+            self.assertIsNone(out)
+
+        with self.subTest(msg="un3"):
+            out = plugin.run(self.un3.to_matrix())
+            self.assertIsNone(out)
+
+    def test_plugin_with_parameters(self):
+        """Test that we can pass parameters to the plugin."""
+
+        plugin = CliffordUnitarySynthesis()
+
+        with self.subTest(msg="min_qubits=2"):
```

Comment:
the following 2 tests are the same, perhaps these can be unified into a single test?